scripts/dataset_processing/ljspeech/extract_ljspeech_phonemes_and_durs.sh

```diff
@@ -0,0 +1,141 @@
+# - <LJSPEECH_BASE>/alignments/wavs/LJ*.TextGrid: MFA output
+# - <LJSPEECH_BASE>/ljspeech_dict.txt: MFA-generated word->phone mapping
+#
+# Note: This script will create a conda environment to set up MFA, so please
+#       make sure that conda is active before running this script.
+#
+# Note: You may need to install OpenBlas yourself if the MFA script can't find
+#       it. (e.g. sudo apt-get install libopenblas-dev)
+#
+# Example Usage:
+# ./extract_ljspeech_phonemes_and_durs.sh \
+#   --train_manifest=/data/manifests/ljspeech_train.json \
+#   --val_manifest=/data/manifests/ljspeech_val.json \
+#   --test_manifest=/data/manifests/ljspeech_test.json \
+#   /data/LJSpeech-1.1
+
+ENV_NAME='aligner'
+
+SAMPLE_RATE=22050
+WINDOW_STRIDE=256
+
+CMUDICT_URL='https://raw.githubusercontent.com/cmusphinx/cmudict/master/cmudict.dict'
+SPLITS_BASE_URL='https://raw.githubusercontent.com/NVIDIA/tacotron2/master/filelists/'
+
+# Usage info
+show_help() {
+cat << EOF
+Usage: $(basename "$0") [-h] \
+          [--skip_env_setup] \
+          [--g2p_dict=<G2P_DICT_PATH>] \
+          <LJSPEECH_BASE>
+Extracts phonemes and their respective durations for the LJSpeech dataset using
+the Montreal Forced Aligner (MFA).
+
+    -h                Help message
+    --skip_env_setup  (Optional) Skips setting up the MFA conda environment
+                      "aligner".  Use only if you already have this set up.
+    --g2p_dict        (Optional) Path to the grapheme to phoneme dictionary
+                      text file, if already generated. If set, skips the G2P
+                      step.
+EOF
+}
+
+SKIP_ENV_SETUP=false
+G2P_DICT=''
+
+TRAIN_MANIFEST=''
+VAL_MANIFEST=''
+TEST_MANIFEST=''
+
+while :; do
+  case $1 in
+    -h|-\?|--help)
+      show_help
+      exit
+      ;;
+    --g2p_dict=?*)
+      G2P_DICT=${1#*=}
+      ;;
+    --skip_env_setup)
+      SKIP_ENV_SETUP=true
+      ;;
+    *)
+      break
+  esac
+  shift
+done
+
+if [[ -z $1 ]]; then
+  echo "Must specify a LJSpeech base directory."
+  exit 1
+fi
+LJSPEECH_BASE=$1
+
+# Check for conda
+read -r -d '' CONDA_MESSAGE << EOM
+This script requires either Anaconda or Miniconda to be active, as it installs the Montreal Forced Aligner via a conda environment.
+See their documentation (https://montreal-forced-aligner.readthedocs.io/en/latest/installation.html) for more details.
+EOM
+conda -h > /dev/null || echo $CONDA_MESSAGE
+
+CONDA_PREFIX=$(conda info --base)
+source $CONDA_PREFIX/etc/profile.d/conda.sh
+
+# Set up env for MFA (env name "aligner") and install
+if $SKIP_ENV_SETUP; then
+  echo "Skipping environment setup. Assuming env name "aligner" exists."
+else
+  echo "Setting up conda environment for MFA (env name \"aligner\")..."
+  conda create -n $ENV_NAME -c conda-forge openblas python=3.8 openfst pynini ngram baumwelch tgt
+  conda activate $ENV_NAME
+  pip install montreal-forced-aligner
+  mfa thirdparty download
+  echo "Conda environment \"$ENV_NAME\" set up."
+fi
+conda activate $ENV_NAME
```

Comment:
Can we fail here as well if conda could not activate? For some reason, it doesn't seem to work inside our container